substrate/frame/revive/src/mock.rs

```diff
@@ -0,0 +1,55 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Helper interfaces and functions, that help with controlling the execution of EVM contracts.
+//! It is mostly used to help with the implementation of foundry cheatscodes for forge test
+//! integration.
+
+use frame_system::pallet_prelude::OriginFor;
+use sp_core::{H160, U256};
+
+use crate::{pallet, DelegateInfo, ExecReturnValue};
+
+/// A trait that provides hooks for mocking EVM contract calls and callers.
+/// This is useful for testing and simulating contract interactions within foundry forge tests.
+pub trait MockHandler<T: pallet::Config> {
```

Comment:
why is that a trait, can't we just use the MockHandlerImpl directly?

Reply:
The  real MockHandlerImpl is implemented here: https://github.com/paritytech/foundry-polkadot/pull/334/files#diff-9e3b7cd881b3ef657db58bf38a66549a3b651f550caa7d623f8d4b20a813ea79R73, having it as a trait gives us a lot more flexibility on how we implement it.

For example some of the mock_calls need to consume the mocks, some don't, I consider that foundry logic having it live in pallet-revive as a concrete type doesn't seem like the right thing for me.